src/main/groovy/edu/oregonstate/mist/locations/frontend/resources/LocationResource.groovy

```diff
@@ -275,6 +292,80 @@ class LocationResource extends Resource {
 
     }
 
+    /**
+     * Turn resultObject to GeoJSON format.
+     *
+     * @param resultObject
+     * @return
+     */
+    private static toGeoJson(ResultObject resultObject) {
+        def geojsonResultObject
+        def ro = resultObject?.data
+
+        if (ro instanceof List) {
+            geojsonResultObject = new GeoFeatureCollection(type: "FeatureCollection")
+            ro.each {
+                geojsonResultObject?.features += adjustGeoFeature(it)
+            }
+        } else {
+            geojsonResultObject = adjustGeoFeature(ro)
+        }
+
+        geojsonResultObject
+    }
+
+    /**
+     * Adjust single resultObject to a Feature object.
+     *
+     * @param ro
+     * @return
+     */
+    private static GeoFeature adjustGeoFeature(ResourceObject ro) {
+        def geojsonResultObject = new GeoFeature(type: "Feature")
+
+        def geoPolygon = null
+        def geoPoint = null
+        def geometry
+
+        // adjust geoPolygon
+        if (ro?.attributes?.geometry?.type && ro?.attributes?.geometry?.coordinates) {
```

Comment:
It's great that you're coding defensively by using `?.` everywhere (something I often forget to do) but if `ro` is null we're just going to end up returning an object full of nulls. I'd almost rather crash in that case.